### PR TITLE
www-servers/nginx: add 1.25.1

### PR DESCRIPTION
**Pull request by hydrapolic**
Compile tested on ~amd64.

```bash
--- nginx-1.24.0-r1.ebuild      2023-05-02 22:22:04.714358183 +0000
+++ nginx-1.25.1.ebuild 2023-06-23 12:14:51.745036472 +0000
@@ -159,7 +159,7 @@
 GEOIP2_MODULE_WD="${WORKDIR}/ngx_http_geoip2_module-${GEOIP2_MODULE_PV}"
 
 # njs-module (https://github.com/nginx/njs, as-is)
-NJS_MODULE_PV="0.7.12"
+NJS_MODULE_PV="5b463b8050377216ad4197cd1e35bb69b35b77e9"
 NJS_MODULE_P="njs-${NJS_MODULE_PV}"
 NJS_MODULE_URI="https://github.com/nginx/njs/archive/${NJS_MODULE_PV}.tar.gz"
 NJS_MODULE_WD="${WORKDIR}/njs-${NJS_MODULE_PV}"
@@ -252,7 +252,7 @@
        stream_javascript
 "
 
-IUSE="aio debug +http +http2 +http-cache libatomic pcre +pcre2 pcre-jit rtmp selinux ssl threads vim-syntax"
+IUSE="aio debug +http +http2 http3 +http-cache libatomic pcre +pcre2 pcre-jit rtmp selinux ssl threads vim-syntax"
 
 for mod in $NGINX_MODULES_STD; do
        IUSE="${IUSE} +nginx_modules_http_${mod}"
@@ -441,6 +441,7 @@
        use aio       && myconf+=( --with-file-aio )
        use debug     && myconf+=( --with-debug )
        use http2     && myconf+=( --with-http_v2_module )
+       use http3     && myconf+=( --with-http_v3_module )
        use libatomic && myconf+=( --with-libatomic )
        use pcre      && myconf+=( --with-pcre --without-pcre2 )
        use pcre-jit  && myconf+=( --with-pcre-jit )
@@ -583,7 +584,7 @@
                myconf+=( --add-module=${HTTP_BROTLI_MODULE_WD} )
        fi
 
-       if use http || use http-cache || use http2 || use nginx_modules_http_javascript; then
+       if use http || use http-cache || use http2 || use http3 || use nginx_modules_http_javascript; then
                http_enabled=1
        fi
```

### CODE REVIEW

**Comment by hydrapolic** [comment]
Hello @ConiKost

**Review by juippis — COMMENTED** [review]
I'm getting some failures due to:
```
cc1: all warnings being treated as errors
```
Could you figure a way to turn that off?

```
src/njs_iterator.c: In function 'njs_object_iterate':
src/njs_iterator.c:358:25: error: storing the address of local variable 'string_obj' in '*args.value' [-Werror=dangling-pointer=[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdangling-pointer=]]
  358 |             args->value = &string_obj;
      |             ~~~~~~~~~~~~^~~~~~~~~~~~~
src/njs_iterator.c:300:58: note: 'string_obj' declared here
  300 |     njs_value_t         *value, *entry, prop, character, string_obj;
      |                                                          ^~~~~~~~~~
src/njs_iterator.c:293:55: note: 'args' declared here
  293 | njs_object_iterate(njs_vm_t *vm, njs_iterator_args_t *args,
      |                                  ~~~~~~~~~~~~~~~~~~~~~^~~~
src/njs_iterator.c: In function 'njs_object_iterate_reverse':
src/njs_iterator.c:522:25: error: storing the address of local variable 'string_obj' in '*args.value' [-Werror=dangling-pointer=[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdangling-pointer=]]
  522 |             args->value = &string_obj;
      |             ~~~~~~~~~~~~^~~~~~~~~~~~~
src/njs_iterator.c:462:58: note: 'string_obj' declared here
  462 |     njs_value_t         *entry, *value, prop, character, string_obj;
      |                                                          ^~~~~~~~~~
src/njs_iterator.c:455:63: note: 'args' declared here
  455 | njs_object_iterate_reverse(njs_vm_t *vm, njs_iterator_args_t *args,
      |                                          ~~~~~~~~~~~~~~~~~~~~~^~~~
cc1: all warnings being treated as errors
make[2]: *** [build/Makefile:700: build/src/njs_iterator.o] Error 1
```

Other than that I'm baffled we don't have a maintainer for nginx o.o

**Comment by cbricart** [comment]
> I'm getting some failures due to:
> 
> ```
> cc1: all warnings being treated as errors
> ```
> 
> Could you figure a way to turn that off?

this has been fixed [upstream](https://github.com/nginx/njs/commit/e60d10f2007c332391d7a104453e13567ebbe52d)

so setting (at least)
`NJS_MODULE_PV="e60d10f2007c332391d7a104453e13567ebbe52d"`
in the ebuild fixes for now, until njs 0.7.13 would be released that has this fix

**Comment by hydrapolic** [comment]
> 

Thanks, I'll update the PR with NJS using a snapshot version.

**Comment by hydrapolic** [comment]
> 

Thanks @juippis, should be ok now.

File: www-servers/nginx/nginx-1.25.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,1067 @@
...
+		--http-fastcgi-temp-path="${EPREFIX}${NGINX_HOME_TMP}"/fastcgi \
+		--http-scgi-temp-path="${EPREFIX}${NGINX_HOME_TMP}"/scgi \
+		--http-uwsgi-temp-path="${EPREFIX}${NGINX_HOME_TMP}"/uwsgi \
+		--with-compat \
+		"${myconf[@]}" || die "configure failed"
+
+	# A purely cosmetic change that makes nginx -V more readable. This can be
+	# good if people outside the gentoo community would troubleshoot and
+	# question the users setup.
+	sed -i -e "s|${WORKDIR}|external_module|g" objs/ngx_auto_config.h || die
```
Comment (juippis) [inline www-servers/nginx/nginx-1.25.1.ebuild:676]:
```
 * SED: the following did not cause any changes
 *     sed -i -e "s|${WORKDIR}|external_module|g" objs/ngx_auto_config.h || die
 * no-op: -e s|/var/tmp/portage/www-servers/nginx-1.25.1/work|external_module|g
```

**Review by juippis — APPROVED** [review]
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/perl5/vendor_perl/5.36/x86_64-linux-thread-multi/auto/nginx/nginx.so

```
Thanks!

File: www-servers/nginx/nginx-1.25.1.ebuild (hunk 1 above)
Comment (juippis) [reply www-servers/nginx/nginx-1.25.1.ebuild:676]: Hmmm I see this only is effective sometimes after configure.

**Comment by hydrapolic** [comment]
Thank @juippis

**Comment by ConiKost** [comment]
> Hello @ConiKost

Sorry :-( I was on vacation and thus wasn't able to look into it.

@juippis thank you for taking care. While nginx has currently has no maintainer, @hydrapolic and me are tying to keep nginx up-to-date :-)

**Comment by cbricart** [comment]
I just saw this got merged to the tree - but it ended up to land in `SLOT=0` rather than intended `SLOT=mainline` 
should I open a bug and an additional PR to fix?

**Comment by ConiKost** [comment]
Ineed. This should be mainline. @hydrapolic Could you fill a PR please.

**Comment by hydrapolic** [comment]
Thanks for noticing, fixed in https://github.com/gentoo/gentoo/pull/31589

**Comment by hydrapolic** [comment]
> 

Hope you enjoyed your holiday :)

**Pull request closed without merging**